third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -0,0 +1,405 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use codespan_reporting::diagnostic::Severity;
+use ethnum::U256;
+use move_model::{
+    ast::{Exp, ExpData, Operation, TempIndex, Value},
+    model::{FunId, FunctionEnv, GlobalEnv, Loc, NodeId, Parameter, QualifiedId},
+    symbol::Symbol,
+    ty::{PrimitiveType, Type},
+};
+use move_stackless_bytecode::{
+    function_target::FunctionData,
+    stackless_bytecode::{AssignKind, AttrId, Bytecode, Constant, Operation as BytecodeOperation},
+};
+use num::ToPrimitive;
+use std::collections::{BTreeMap, BTreeSet, VecDeque};
+
+// ======================================================================================
+// Entry
+
+/// Generate code for the given function from it's AST representation in the env.
+/// This returns `FunctionData` suitable for the bytecode processing pipeline.
+pub fn generate_bytecode(env: &GlobalEnv, fid: QualifiedId<FunId>) -> FunctionData {
+    let func_env = env.get_function(fid);
+    let mut gen = Generator {
+        func_env,
+        temps: Default::default(),
+        scopes: Default::default(),
+        attr_table: Default::default(),
+        code: vec![],
+    };
+    let mut scope = BTreeMap::new();
+    for Parameter(name, ty) in gen.func_env.get_parameters() {
+        let temp = gen.new_temp(ty);
+        scope.insert(name, temp);
+    }
+    let mut results = vec![];
+    for ty in gen.func_env.get_result_type().flatten() {
+        let temp = gen.new_temp(ty);
+        results.push(temp)
+    }
+    gen.scopes.push_back(scope);
+    if let Some(def) = gen.func_env.get_def().cloned() {
+        // Need to clone expression if present because of sharing issues with `gen`. However, because
+        // of internalization, clone is cheap.
```

Comment:
I think "interning" is better than "internalization" here, which has too many other meanings even if it might be used in this case (which isn't impossible).